controllers/rabbitmq/transporturl_controller.go

```diff
@@ -0,0 +1,192 @@
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	helper "github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/secret"
+	oko_secret "github.com/openstack-k8s-operators/lib-common/modules/common/secret"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	rabbitmqv1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/rabbitmq/v1beta1"
+	"github.com/openstack-k8s-operators/openstack-operator/pkg/rabbitmq"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// GetClient -
+func (r *TransportURLReconciler) GetClient() client.Client {
+	return r.Client
+}
+
+// GetKClient -
+func (r *TransportURLReconciler) GetKClient() kubernetes.Interface {
+	return r.Kclient
+}
+
+// GetLogger -
+func (r *TransportURLReconciler) GetLogger() logr.Logger {
+	return r.Log
+}
+
+// GetScheme -
+func (r *TransportURLReconciler) GetScheme() *runtime.Scheme {
+	return r.Scheme
+}
+
+// TransportURLReconciler reconciles a TransportURL object
+type TransportURLReconciler struct {
+	client.Client
+	Kclient kubernetes.Interface
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=transporturls,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=transporturls/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=transporturls/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rabbitmq.com,resources=rabbitmqclusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete;
+
+// Reconcile - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *TransportURLReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// Fetch the TransportURL instance
+	instance := &rabbitmqv1beta1.TransportURL{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Lookup RabbitmqCluster instance by name
+	labelSelector := map[string]string{
+		"namespace": instance.Namespace,
+		"name":      instance.Spec.RabbitmqClusterName,
+	}
+	rabbit, err := rabbitmq.GetRabbitmqCluster(ctx, helper, instance, labelSelector)
```

Comment:
Should we add a `TODO` to reflect that at some point we should actually watch the `rabbitmqclusters.rabbitmq.com` CRs so that if the change we can update the `transporturls.rabbitmq.openstack.org` accordingly.

Reply:
done